src/sql/workbench/common/views.ts

```diff
@@ -22,6 +22,8 @@ export interface ITreeItem extends vsITreeItem {
 	childProvider?: string;
 	payload?: IConnectionProfile; // its possible we will want this to be more generic
 	sqlIcon?: string;
+	type?: string;
+	nodeInfo?: NodeInfo;
```

Comment:
Added NodeInfo to make it similar to ObjectExplorerContext

Reply:
where is this used? I'd rather not change a core object like this unless necessary.

Reply:
It's used for all scripting actions and extension actions such as Properties, Generate Scripts etc

Reply:
I think it makes more sense to just have a method on the shim service for getting this object, and have those actions grab it directly from the shim service.